Dockerfile

```diff
@@ -0,0 +1,51 @@
+##############################################################
+# This file is intended to be used with ./docker-compose.yml #
+##############################################################
+
+FROM node:10.14.1-alpine as build
+# working directory
+WORKDIR /usr/src/app
+
+# global environment setup : yarn + dependencies needed to support node-gyp
+RUN apk --no-cache --virtual add \
+    python \
+    make \
+    g++ \
+    yarn 
+
+# copy just the dependency files and configs needed for install
+COPY packages/peregrine/package.json ./packages/peregrine/package.json
+COPY packages/pwa-buildpack/package.json ./packages/pwa-buildpack/package.json
+COPY packages/upward-js/package.json ./packages/upward-js/package.json
+COPY packages/upward-spec/package.json ./packages/upward-spec/package.json
+COPY packages/venia-concept/package.json ./packages/venia-concept/package.json
+COPY package.json yarn.lock babel.config.js browserslist.js ./
+
+# install dependencies with yarn
+RUN yarn install
+
```

Comment:
The idea behind copying over all the package.json dependency files independently is to utilize Docker's [cache](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache) so that the build is fast on subsequent builds. The idea is that you only add files needed for the build, then run a build, then add all the other files because, presumably, dependencies change less than other files and the cache will not be disrupted until there is a change in a file. So the more files in a cached layer, the more likely that there will be changes and therefore require a rebuild of that layer.

When an item is added to a post-install then it will need to be added here too. Right now we don't have any other dependencies needed for the build other than the files above the install line.

Reply:
Is there a reason to run "yarn install" and copy these files in Dockerfile build phase at all? Perhaps at least for development environment this could be done on entrypoint script, so that image itself does not need to contain the application files.

Usually application files are only "burned" to the image when final deployable production image is created.

Reply:
in order to build you need to `yarn install`. without the build dependencies then it would throw errors when running `yarn run build`. above I have copied only the build dependencies, package.json and relevant files needed for the build. Below on line 28 I am copying over the remainder of the package files before building. All files in the repo are needed to create a build that can then be spun up with the `CMD`. 

If you look at what `yarn run build` is executing, it runs the build commands of all the directories in the workspace under `packages/`, therefore it is needed to run `yarn install` to get all relevant dependencies needed for the build scripts to execute, as well as copy over all the contents of each package.